src/main/java/mate/academy/hibernate/relations/service/impl/CountryServiceImpl.java

```diff
@@ -1,16 +1,25 @@
 package mate.academy.hibernate.relations.service.impl;
 
+import java.util.Optional;
+import mate.academy.hibernate.relations.dao.CountryDao;
 import mate.academy.hibernate.relations.model.Country;
 import mate.academy.hibernate.relations.service.CountryService;
 
 public class CountryServiceImpl implements CountryService {
+    private final CountryDao dao;
+
+    public CountryServiceImpl(CountryDao dao) {
+        this.dao = dao;
+    }
+
     @Override
     public Country add(Country country) {
-        return null;
+        return dao.add(country);
     }
 
     @Override
     public Country get(Long id) {
-        return null;
+        Optional<Country> result = dao.get(id);
+        return result.orElseGet(Country::new);
```

Comment:
very  confusing logic... if our entity does not exists, your code would create new. but it's not what we asked for by id. consider using `orElseThrow()`